checkST2110.js

```diff
@@ -971,210 +1164,481 @@ const test_30_62_4 = (sdp, params) => {
       }
       // TODO not clear if this has to be an integer number of microseconds or can be a decimal value?
     }
   }
+  if (params.verbose && errors.length == 0) {
+    console.log('Test Passed: ST 2110-21 Section 8.2 Test 1 - TROFF parameter is an acceptable value ');
+  }
   return errors;
 };
 
-// Test ST 2110-21 Section 8.2 - When traffic shaping and CMAX parameter specified, it is an acceptable value
+// ST 2110-21 Section 8.2 Test 2 - When traffic shaping and CMAX parameter specified, it is an acceptable value
 const test_21_82_2 = (sdp, params) => {
   if (params.shaping === false || params.audioOnly === true) {
+    if (params.verbose) {
+      console.log('Test Skipped: ST 2110-21 Section 8.2 Test 2 - When CMAX parameter specified it is an acceptable value. Use --shaping to test.');
+    }
     return [];
   }
-  let [ mtParams, errors ] = extractMTParams(sdp, params);
-  for ( let stream of mtParams ) {
-    if ( typeof stream.CMAX !== 'undefined' ) {
+  let [mtParams, errors] = extractMTParams(sdp, params);
+  for (let stream of mtParams) {
+    if (typeof stream.CMAX !== 'undefined') {
       let cmax = +stream.CMAX;
       if (isNaN(cmax)) {
-        errors.push(new Error(`Line ${stream._line}: For stream ${stream._streamNumber}, CMAX parameter is not a number, as per SMPTE ST 2110-21 Section 8.2.`));
+        errors.push(new Error(`Line ${stream._line}: For stream ${stream._streamNumber}, CMAX parameter is not a number, as per ST 2110-21 Section 8.2.`));
         continue;
       }
       if (Number.isInteger(cmax) === false) {
-        errors.push(new Error(`Line ${stream._line}: For stream ${stream._streamNumber}, CMAX parameter in not an integer, as per SMPTE ST 2110-21 Section 8.2.`));
+        errors.push(new Error(`Line ${stream._line}: For stream ${stream._streamNumber}, CMAX parameter in not an integer, as per ST 2110-21 Section 8.2.`));
       }
       if (cmax < 1) {
-        errors.push(new Error(`Line ${stream._line}: For stream ${stream._streamNumber}, CMAX parameter makes no sense unless it is a positive value, as per SMPTE ST 2110-21 Section 8.2.`));
+        errors.push(new Error(`Line ${stream._line}: For stream ${stream._streamNumber}, CMAX parameter makes no sense unless it is a positive value, as per ST 2110-21 Section 8.2.`));
+      }
+    }
+    else {
+      if (params.verbose) {
+        console.log('Test Skipped: ST 2110-21 Section 8.2 Test 2 - CMAX Not Present.');
       }
     }
   }
+  if (params.verbose && errors.length == 0) {
+    console.log('Test Passed: ST 2110-21 Section 8.2 Test 2 - When traffic shaping and CMAX parameter specified, it is an acceptable value.');
+  }
   return errors;
 };
 
+const typesPermitted22 = ['2110TPN', '2110TPNL', '2110TPW'];
+
+// ST 2110-22 Section 5.3 Test 1 - Traffic Shaping and Delivery Timing - TP parameter is an acceptable value
+const test_22_53_1 = (sdp, params) => {
```

Comment:
It seems to me that this `test_22_53_1` is basically a duplicate of `test_21_81_1` and `test_21_81_2` (at least now that ST 2110-22:2022 has made `2110TPNL` permitted...)

If we don't want to do any further rationalization at this point, let's at least point it out:

```suggestion
// ST 2110-22 Section 5.3 Test 1 - Traffic Shaping and Delivery Timing - TP parameter is an acceptable value
// Note: this is almost identical to test_21_81_2 used for ST 2110-20
const test_22_53_1 = (sdp, params) => {
```

Reply:
agreed

Reply:
Committed